freeipa/src/main/java/com/sequenceiq/freeipa/converter/telemetry/TelemetryConverter.java

```diff
@@ -0,0 +1,131 @@
+package com.sequenceiq.freeipa.converter.telemetry;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.common.api.cloudstorage.old.AdlsGen2CloudStorageV1Parameters;
+import com.sequenceiq.common.api.cloudstorage.old.S3CloudStorageV1Parameters;
+import com.sequenceiq.common.api.telemetry.model.Features;
+import com.sequenceiq.common.api.telemetry.model.Logging;
+import com.sequenceiq.common.api.telemetry.model.Telemetry;
+import com.sequenceiq.common.api.telemetry.request.LoggingRequest;
+import com.sequenceiq.common.api.telemetry.request.TelemetryRequest;
+import com.sequenceiq.common.api.telemetry.response.FeaturesResponse;
+import com.sequenceiq.common.api.telemetry.response.LoggingResponse;
+import com.sequenceiq.common.api.telemetry.response.TelemetryResponse;
+import com.sequenceiq.common.api.type.FeatureSetting;
+
+@Component
+public class TelemetryConverter {
+
+    private final boolean freeIpaTelemetryEnabled;
+
+    private final boolean reportDeplymentLogs;
+
+    private final String databusEndpoint;
+
+    public TelemetryConverter(@Value("${freeipa.telemetry.enabled:false}") boolean freeIpaTelemetryEnabled,
+            @Value("${cluster.deployment.logs.report:false}") boolean reportDeplymentLogs,
+            @Value("${altus.databus.endpoint:}") String databusEndpoint) {
+        this.freeIpaTelemetryEnabled = freeIpaTelemetryEnabled;
+        this.reportDeplymentLogs = reportDeplymentLogs;
+        this.databusEndpoint = databusEndpoint;
+    }
+
+    public Telemetry convert(TelemetryRequest request) {
+        Telemetry telemetry = null;
+        if (freeIpaTelemetryEnabled && request != null) {
+            telemetry = new Telemetry();
+            LoggingRequest loggingRequest = request.getLogging();
+            Logging logging = createLoggingFromRequest(loggingRequest);
+            telemetry.setLogging(logging);
+            Features features = null;
+            features = createFeaturesFromRequest(request, features);
+            telemetry.setFeatures(features);
```

Comment:
Variable not necessary, can be inlined in setter invocation. Does it make sense to pass in `null` to `createFeaturesFromRequest()`? Why not initialize the default return value inside that method?